src/mgr/DaemonServer.cc

```diff
@@ -562,6 +562,7 @@ bool DaemonServer::handle_report(const ref_t<MMgrReport>& m)
       auto priv = m->get_connection()->get_priv();
       auto session = static_cast<MgrSession*>(priv.get());
       if (!session) {
+        lock.unlock();
```

Comment:
could you take this opportunity to use a `unique_lock<>` ?

Reply:
ack